staging/src/k8s.io/dynamic-resource-allocation/deviceclass/extendedresourcecache/extendedresourcecache_test.go

```diff
@@ -182,28 +246,78 @@ func TestExtendedResourceCache(t *testing.T) {
 		t.Error("Expected default mapping for gpu-class")
 	}
 
+	// deviceClass3 is older than deviceClass1, hence it won't replace deviceClass1
+	_, err = client.ResourceV1().DeviceClasses().Create(tCtx, deviceClass3, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create device class: %v", err)
+	}
+	time.Sleep(1 * time.Second)
+
+	// should keep deviceClass1, since it is newer than deviceClass3
+	deviceClassName = cache.GetDeviceClass("example.com/gpu")
+	if deviceClassName != "gpu-class" {
+		t.Errorf("Expected to find device class 'gpu-class' for 'example.com/gpu', got %s", deviceClassName)
+	}
+
+	// deviceClass4 is newer than deviceClass1, hence it will replace deviceClass1
+	_, err = client.ResourceV1().DeviceClasses().Create(tCtx, deviceClass4, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create device class: %v", err)
+	}
+	time.Sleep(1 * time.Second)
+
+	// deviceClass4 replaces deviceClass1, since it is newer with the same example.com/gpu extended resource name
+	deviceClassName = cache.GetDeviceClass("example.com/gpu")
+	if deviceClassName != "gpu-class-4" {
+		t.Errorf("Expected to find device class 'gpu-class' for 'example.com/gpu', got %s", deviceClassName)
+	}
+
+	// deviceClass0 is created at the same time as deviceClass4, but its name is alphabetically ordered earlier,
+	//  hence it will replace deviceClass4
+	_, err = client.ResourceV1().DeviceClasses().Create(tCtx, deviceClass0, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create device class: %v", err)
+	}
+	time.Sleep(1 * time.Second)
+
+	// deviceClass0 replaces deviceClass4, it is created at the same time as deviceClass4, but its name is
+	// alphabetically ordered earlier
+	deviceClassName = cache.GetDeviceClass("example.com/gpu")
+	if deviceClassName != "gpu-class-0" {
+		t.Errorf("Expected to find device class 'gpu-class' for 'example.com/gpu', got %s", deviceClassName)
+	}
```

Comment:
Please also add test cases for implicit vs. explicit clashes.

Reply:
i think they cannot clash, see above comment.